### PR TITLE
Add flex-shrink, flex-basis and flex shorthand utilities

### LINKED ISSUE #6: Specificity problem
If I have flat selectors and I need override some flex declarations, there is specificity problem: https://github.com/suitcss/utils-flex/blob/master/lib/flex.css#L144-L147

Any suggestions about that?

Thanks

### PR DESCRIPTION
**Pull request by simonsmith**
A mixture of things here:

* Fix #6 by removing the IE10 hack, adding `flex-shrink` and documenting the usage. The test page also doubles up as a guide.
* Add `u-flexBasisAuto` and `u-flexBasisZero`. These work well as overrides as we set `flex-basis: 0%` in a lot of places
* Add `u-flexNone`, `u-flexInitial` and `u-flexAuto`. More useful overrides.

@giuseppeg @illycz

### CODE REVIEW

**Comment by simonsmith** [comment]
@timkelty Do you have any thoughts too? I know you've been using this util in the past :)

File: lib/flex-lg.css [hunk 1]
```diff
@@ -256,5 +274,50 @@
...
-}
+  /**
+   * Basis
+   */
+
+  .u-lg-flexBasisAuto {
+    flex-basis: auto !important;
+  }
+
+  .u-lg-flexBasisZero {
```
Comment (giuseppeg) [inline lib/flex-lg.css]: why Zero instead of 0?

File: lib/flex-lg.css [hunk 2]
```diff
@@ -256,5 +274,50 @@
...
+    flex-basis: 0% !important;
```
Comment (giuseppeg) [inline lib/flex-lg.css]: is the unit required?

File: lib/flex-lg.css (hunk 1 above)
Comment (simonsmith) [inline lib/flex-lg.css]: I guess I was copying `Auto` and `Initial`. It can be `0` though.

File: lib/flex-lg.css (hunk 2 above)
Comment (simonsmith) [inline lib/flex-lg.css]: Hmm, perhaps not. It's needed to fix IE10 but only when using the `flex` shorthand. I'll take it away and browser test.

**Comment by simonsmith** [comment]
Addressed comments and added some tests, which caught a minor issue with having to set `flex-basis` explicitly in IE10.

**Comment by timkelty** [comment]
@simonsmith looks good at first glance, but will review later today.

**Comment by timkelty** [comment]
Love it, love all of it. You all are beautiful! :shipit:

**Pull request merged**